dev/io.openliberty.org.jboss.resteasy.common/src/org/jboss/resteasy/client/jaxrs/engines/ManualClosingApacheHttpClient43Engine.java

```diff
@@ -123,39 +122,56 @@ public String run() throws Exception
 
    public ManualClosingApacheHttpClient43Engine()
    {
-      this.httpClient = createDefaultHttpClient();
-      this.allowClosingHttpClient = true;
+      this(null, null, true, null);
    }
 
    public ManualClosingApacheHttpClient43Engine(final HttpHost defaultProxy)
    {
-      this.defaultProxy = defaultProxy;
-      this.httpClient = createDefaultHttpClient();
-      this.allowClosingHttpClient = true;
+      this(null, null, true, defaultProxy);
    }
 
    public ManualClosingApacheHttpClient43Engine(final HttpClient httpClient)
    {
-      this.httpClient = httpClient;
-      this.allowClosingHttpClient = true;
+      this(httpClient, null, true, null);
    }
 
    public ManualClosingApacheHttpClient43Engine(final HttpClient httpClient, final boolean closeHttpClient)
    {
-      if (closeHttpClient && !(httpClient instanceof CloseableHttpClient))
-      {
-         throw new IllegalArgumentException(
-               "httpClient must be a CloseableHttpClient instance in order for allowing engine to close it!");
-      }
-      this.httpClient = httpClient;
-      this.allowClosingHttpClient = closeHttpClient;
+      this(httpClient, null, closeHttpClient, null);
    }
 
    public ManualClosingApacheHttpClient43Engine(final HttpClient httpClient, final HttpContextProvider httpContextProvider)
    {
+      this(httpClient, httpContextProvider, true, null);
+   }
+
+   private ManualClosingApacheHttpClient43Engine(HttpClient httpClient, final HttpContextProvider httpContextProvider, final boolean closeHttpClient, HttpHost defaultProxy)
+   {
+       if (httpClient == null) {
+           httpClient = createDefaultHttpClient();
+       }
+       if (closeHttpClient && !(httpClient instanceof CloseableHttpClient))
+       {
+          throw new IllegalArgumentException(
+                "httpClient must be a CloseableHttpClient instance in order for allowing engine to close it!");
+       }
       this.httpClient = httpClient;
       this.httpContextProvider = httpContextProvider;
-      this.allowClosingHttpClient = true;
+      this.allowClosingHttpClient = closeHttpClient;
+      this.defaultProxy = defaultProxy;
+      
+      try {
+          int threshold = Integer.parseInt(
+              ResteasyConfigProvider.getConfig()
```

Comment:
`ResteasyConfigProvider` is deprecated. Is there a reason not to use `ConfigurationFactory.getInstance().getConfiguration()`?